subsys/sd/mmc.c

```diff
@@ -422,6 +460,88 @@ static int mmc_set_timing(struct sd_card *card, struct mmc_ext_csd *ext)
 		return -ENOTSUP;
 	}
 
+	/* Set card timing mode */
```

Comment:
The flow here is a bit confusing to me: It seems that the paths this function can take are the following:
- 26MHz is best speed supported: return `-ENOTSUP`
- 52MHz is best speed supported: call `mmc_set_HS_timing` and return
- 200MHz is best speed supported: call `mmc_set_HS_timing` and continue into function

I feel that for 200MHz, calling a function like `mmc_set_hs200_timing` would make more sense. For HS400, a function could be implemented called `mmc_set_hs400_timing`, which itself would call `mmc_set_hs_timing`, and `mmc_set_hs200_timing`. I'll leave this up to you though.

Reply:
hmmmm.... it appears that a git rebase completely fubar'ed this function. It did not look like this at all when I wrote it

Reply:
Actually that's not correct, it returns 0 when 26 MHz is best supported, and returns ENOTSUP if no supported mode is encoded in EXT_CSD (unlikely)